user-service/src/main/java/rs/edu/raf/banka/user_service/model/AbstractUser.java

```diff
@@ -0,0 +1,62 @@
+package rs.edu.raf.banka.user_service.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.lang.Nullable;
+
+import javax.persistence.*;
+
+@MappedSuperclass
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public abstract class AbstractUser {
```

Comment:
Mislim da apstrakcija preivše komplikuje. Sasvim je dovoljno da se doda nova rola, kao što ima ROLE_GL_ADMIN i ROLE_ADMIN, tako da bude i ROLE_SUPERVISOR i ROLE_AGENT.

Reply:
Za role supervisor se slažem, ali agent ima 3 dodatne kolone u odnosu na običnog usera ili nekog drugog u sistemu. Ovakva apstrakcija (extendovanje) je urađena i na berza servisu takođe da nam ne bi svaki tip hartija od vrijednosti imao ista sva polja od ostalih. Ne znam da li će još neke role da postoje, ali ne bi bilo zgodno da svaki user u sistemu ima sva moguća polja svih rola gdje su mu određena null u zavisnosti od role. Nije problem da se vrati na staro pa da nam svaki user ima ta tri opciona polja.